lib/resty/healthcheck.lua

```diff
@@ -308,12 +314,14 @@ function checker:add_target(ip, port, hostname, is_healthy, hostheader)
     end
 
     -- target does not exist, go add it
-    target_list[#target_list + 1] = {
-      ip = ip,
-      port = port,
-      hostname = hostname,
-      hostheader = hostheader,
-    }
+    if not found then
+      target_list[#target_list + 1] = {
+        ip = ip,
+        port = port,
+        hostname = hostname,
+        hostheader = hostheader,
+      }
+    end
     target_list = serialize(target_list)
```

Comment:
what's the expected behaviour if the target is found and not marked for removal? should any of the following be any different?  it seems the `target_list` array isn't modified, but it's re-serialized and the event is raised.

Reply:
Nice catch, thanks! I added a check: if `purge_time` is `nil`, do nothing.